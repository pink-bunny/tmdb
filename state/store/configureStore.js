import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import reducer from './reducer';

const loggerMiddleware = (store) => (next) => (action) => {
  /* eslint-disable no-console */
  console.group(action.type);
  console.info('dispatching', action);
  const result = next(action);
  console.log('next state', store.getState());
  console.groupEnd();
  /* eslint-enable */

  return result;
};

export default function configureStore() {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(reducer, undefined, composeWithDevTools(middlewareEnhancer));

  return store;
}
