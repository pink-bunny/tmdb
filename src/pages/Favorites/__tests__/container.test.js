import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import ConnectedFavorites from '../container';

jest.mock('../../../../state/favorites/selectors', () => ({
  isFavoritesLoading: jest.fn(() => false),
  favoritesTotalItems: jest.fn(() => 21),
  favoritesCurrentPage: jest.fn(() => 1),
  favoritesError: jest.fn(() => ''),
  favoritesList: jest.fn(() => [{
    id: 1,
    title: 'Title',
    overview: 'Overview',
    poster_path: 'Path' /* eslint-disable-line camelcase */
  }])
}));

describe('Favorites container', () => {
  const store = configureStore()({});
  store.dispatch = jest.fn();
  const wrapper = shallow(<ConnectedFavorites store={store} />);
  const container = wrapper.dive().dive();

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
