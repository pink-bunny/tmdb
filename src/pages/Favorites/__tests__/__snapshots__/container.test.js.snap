// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites container matches snapshot 1`] = `
<Favorites
  currentPage={1}
  error=""
  fetchList={[Function]}
  list={
    Array [
      Object {
        "id": 1,
        "overview": "Overview",
        "poster_path": "Path",
        "title": "Title",
      },
    ]
  }
  loading={false}
  removeModalAction={[Function]}
  totalItems={21}
/>
`;
